diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7c22bb77..d633ab0e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -109,14 +109,14 @@ ENDIF()
 #find_package( ScaLAPACK REQUIRED)
 
 IF(NOT USE_EXTERNAL_ELPA)
-  TARGET_LINK_LIBRARIES(elsi PRIVATE elpa)
+  TARGET_LINK_LIBRARIES(elsi PUBLIC elpa)
 ENDIF()
 IF(NOT USE_EXTERNAL_OMM)
-  TARGET_LINK_LIBRARIES(elsi PRIVATE OMM MatrixSwitch)
+  TARGET_LINK_LIBRARIES(elsi PUBLIC OMM MatrixSwitch)
 ENDIF()
 IF(ENABLE_PEXSI)
   IF(NOT USE_EXTERNAL_PEXSI)
-    TARGET_LINK_LIBRARIES(elsi PRIVATE
+    TARGET_LINK_LIBRARIES(elsi PUBLIC
       pexsi
       superlu_dist
       ptscotchparmetis
@@ -128,17 +128,17 @@ IF(ENABLE_PEXSI)
   ENDIF()
 ENDIF()
 IF(ENABLE_SIPS)
-  TARGET_LINK_LIBRARIES(elsi PRIVATE sips)
+  TARGET_LINK_LIBRARIES(elsi PUBLIC sips)
 ENDIF()
 IF(NOT USE_EXTERNAL_NTPOLY)
-  TARGET_LINK_LIBRARIES(elsi PRIVATE NTPoly)
+  TARGET_LINK_LIBRARIES(elsi PUBLIC NTPoly)
 ENDIF()
 IF(ENABLE_BSEPACK)
   IF(NOT USE_EXTERNAL_BSEPACK)
-    TARGET_LINK_LIBRARIES(elsi PRIVATE bsepack sseig)
+    TARGET_LINK_LIBRARIES(elsi PUBLIC bsepack sseig)
   ENDIF()
 ENDIF()
-TARGET_LINK_LIBRARIES(elsi PRIVATE
+TARGET_LINK_LIBRARIES(elsi PUBLIC
   fortjson
   ${LIBS})
 
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index a444127f..49156cd1 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -26,38 +26,6 @@ MACRO(add_test_target arg1)
   ADD_DEPENDENCIES(${arg1} elsi)
 
   TARGET_LINK_LIBRARIES(${arg1} PRIVATE elsi)
-  IF(NOT USE_EXTERNAL_OMM)
-    TARGET_LINK_LIBRARIES(${arg1} PRIVATE OMM MatrixSwitch)
-  ENDIF()
-  IF(NOT USE_EXTERNAL_ELPA)
-    TARGET_LINK_LIBRARIES(${arg1} PRIVATE elpa)
-  ENDIF()
-  TARGET_LINK_LIBRARIES(${arg1} PRIVATE fortjson)
-  IF(ENABLE_SIPS)
-    TARGET_LINK_LIBRARIES(${arg1} PRIVATE sips)
-  ENDIF()
-  IF(ENABLE_PEXSI)
-    IF(NOT USE_EXTERNAL_PEXSI)
-      TARGET_LINK_LIBRARIES(${arg1} PRIVATE
-        pexsi
-        superlu_dist
-        ptscotchparmetis
-        ptscotch
-        ptscotcherr
-        scotchmetis
-        scotch
-        scotcherr)
-    ENDIF()
-  ENDIF()
-  IF(NOT USE_EXTERNAL_NTPOLY)
-    TARGET_LINK_LIBRARIES(${arg1} PRIVATE NTPoly)
-  ENDIF()
-  IF(ENABLE_BSEPACK)
-    IF(NOT USE_EXTERNAL_BSEPACK)
-      TARGET_LINK_LIBRARIES(${arg1} PRIVATE bsepack sseig)
-    ENDIF()
-  ENDIF()
-  TARGET_LINK_LIBRARIES(${arg1} PRIVATE ${LIBS})
 
   TARGET_INCLUDE_DIRECTORIES(${arg1} PRIVATE
     ${INC_PATHS}
